kcidb/create_object.py

```diff
@@ -59,101 +59,43 @@ def get_kernel_hash(kernel_version):
 
 base_kernel_hash=str(get_kernel_hash(kernel_version).stdout).strip("\n")
 
-json_template = """
-{
-    "revisions": [
-        {
-            "id": "2c85ebc57b3e1817b6ce1a6b703928e113a90442",
-            "origin": "gkernelci",
-            "git_repository_url": "git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git",
-            "git_repository_branch": "5.8",
-            "contacts": [
-                "Mike Pagano <mpagano@gentoo.org>"
+r_id = str(base_kernel_hash) + "+" + str(patchset_hash)
+print(r_id)
+
+data = dict(
+    version=dict(
+        major=3,
+        minor=0
+    ),
+    revisions=[
+        dict(
+            id=r_id,
+            origin="gkernelci",
+            git_repository_url="git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git",
+            git_repository_branch=kernel_version,
+            contacts=[
+                "Mike Pagano <mpagano@gentoo.org>",
+                "Alice Ferrazzi <alicef@gentoo.org>"
             ],
-            "patch_mboxes": [
-                {
-                    "name": "1500_XATTR_USER_PREFIX.patch",
-                    "url": "https://gitweb.gentoo.org/proj/linux-patches.git/plain/1500_XATTR_USER_PREFIX.patch?h=5.8&id=1d996290ed3f1ffdc4ea2d9b4c4d2cf19ccc77d3"
-                },
-                {
-                    "name": "1510_fs-enable-link-security-restrictions-by-default.patch",
-                    "url": "https://gitweb.gentoo.org/proj/linux-patches.git/plain/1510_fs-enable-link-security-restrictions-by-default.patch?h=5.8&id=1d996290ed3f1ffdc4ea2d9b4c4d2cf19ccc77d3"
-                }
-            ],
-            "log_url": "https://kernel-ci.emjay-embedded.co.uk/api/v2/logs/16852/raw",
-            "valid": true
-        }
-    ],
-    "builds": [
-        {
-            "id": "gkernelci:50-6",
-            "origin": "gkernelci",
-            "revision_id": "9ece50d8a470ca7235ffd6ac0f9c5f0f201fe2c8+bf3c6a7247ae590c0d2965622961b74b6c99a92fec70d07fa4025cb6fcb944a9",
-            "architecture": "amd64",
-            "log_url": "https://kernel-ci.emjay-embedded.co.uk/api/v2/logs/16853/raw",
-            "valid": true,
-            "misc": {
-                "url": "https://kernel-ci.emjay-embedded.co.uk/#/builders/50/builds/6"
-            }
-        }
+            patch_mboxes=get_patches_list(),
+            log_url="https://kernel-ci.emjay-embedded.co.uk/api/v2/logs/16852/raw",
+            valid=args.result,
```

Comment:
This seems to be using the same value as the build does. However, revision's `valid` should indicate whether you successfully applied patches or not, and only that. And if patches failed to apply, then there should not be a build object.

Do you think you could fix that?

Reply:
I see. I didn't know the difference. I will fix it :+1:

Reply:
Awesome, thanks :grinning: